### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by wasifhossain**

### CODE REVIEW

File: lib/active_model/serializer.rb [hunk 1]
```diff
@@ -73,7 +73,7 @@ def self.serializer_lookup_chain_for(klass, namespace = nil)
     # Used to cache serializer name => serializer class
     # when looked up by Serializer.get_serializer_for.
     def self.serializers_cache
-      @serializers_cache ||= ThreadSafe::Cache.new
+      @serializers_cache ||= Concurrent::Map.new
```
Comment (bf4) [inline lib/active_model/serializer.rb]: kind of odd.  I'm not sure where the dependency is coming from.  I know later versions of Rails introduce concurrent-ruby.  I guess thread_safe was coming from active_support?

File: lib/active_model/serializer/reflection.rb [hunk 2]
```diff
@@ -87,8 +87,8 @@ def initialize(*)
       #       meta ids: ids
       #     end
       #   end
-      def link(name, value = nil)
-        options[:links][name] = block_given? ? Proc.new : value
+      def link(name, value = nil, &block)
+        options[:links][name] = block_given? ? block : value
```
Comment (bf4) [inline lib/active_model/serializer/reflection.rb]: I'm install 2.7 locally so I can figure out what you're seeing. As I wrote in the other PR, this change doesn't make sense to me

Comment (bf4) [reply lib/active_model/serializer/reflection.rb]: ok, this makes more sense
> lib/active_model/serializer/reflection.rb:91: warning: Capturing the given block using Proc.new is
deprecated; use `&block` instead

Comment (bf4) [reply lib/active_model/serializer/reflection.rb]: per https://github.com/ruby/ruby/commit/9f1fb0a17febc59356d58cef5e98db61a3c03550

File: lib/active_model/serializer.rb (hunk 1 above)
Comment (wasifhossain) [reply lib/active_model/serializer.rb]: `thread_safe` is needed by 2 different sources:
- `active_support`
- `grape` (through `virtus`)

**Pull request merged**